qualtran/bloqs/rotations/hamming_weight_phasing.py

```diff
@@ -168,7 +175,10 @@ def build_composite_bloq(
         x, junk, out = bb.add(HammingWeightCompute(self.bitsize), x=x)
         out, phase_grad = bb.add(
             AddScaledValIntoPhaseReg(
-                self.phase_oracle.cost_dtype, self.b_grad, self.exponent / 2, self.gamma_bitsize
+                self.phase_oracle.cost_dtype.bitsize,
+                self.b_grad,
+                (self.exponent / 2) * (2 ** self.bitsize.bit_length()),
+                self.gamma_bitsize,
```

Comment:
keyword arguments would not be amiss here

Reply:
added